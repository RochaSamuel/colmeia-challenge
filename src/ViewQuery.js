import React, { Component } from 'react'

const View = (props) => {
    const cards = props.params.professores.map((professor, index) => {
        if (props.params.search.toLowerCase() === professor.nome.toLowerCase().substr(0, props.params.search.length)) {
            return (
                <div className="col s6" key={index}>
                    <div className="card grey lighten-5">
                        <div className="card-content black-text">
                            <div className="top-container">
                                <div className="left-container">
                                    <div className="professor-img ">
                                        <img src={professor.imagem._url} alt="" className="circle img" />
                                    </div>
                                </div>
                                <div className="middle-container">
                                    <span className="card-title">{professor.nome}</span>
                                    <p>Disciplinas:</p>
                                    {professor.materia.map((item, index) => {
                                        return (<div key={index} className="professor-disciplina">{item}</div>);
                                    })}
                                </div>
                                <div className="right-container">
                                    <div className="professor-nota">{professor.nota}<i className="tiny material-icons">star</i></div>
                                </div>
                            </div>
                            <div className="curriculo-title">Curr√≠culo</div>
                            <p>{professor.curriculo}</p>
                        </div>
                        <div className="card-action">
                            <div className="top-container">
                                <div className="left-container"><i className="tiny material-icons icon">location_on</i>{professor.bairro}</div>
                                <div className="right-container">
                                    <a className="button" href="/">Selecionar</a>
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
            );
        } else {
            return <div key={index}></div>;
        }
    });
    return (
        <div className="row">{cards}</div>
    );
}


class ViewQuery extends Component {
    render() {
        return (
            <View params={this.props.params} />
        );
    }
}

export default ViewQuery