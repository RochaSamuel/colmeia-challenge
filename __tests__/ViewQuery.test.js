import React from 'react';
import { shallow } from 'enzyme';
import ViewQuery from '../src/ViewQuery';

describe('Testing ViewQuery Component', () => {
    it('should render correctly', () => {
        const wrapper = shallow(
            <ViewQuery />
        );

        expect(wrapper).toMatchSnapshot();
    });
});