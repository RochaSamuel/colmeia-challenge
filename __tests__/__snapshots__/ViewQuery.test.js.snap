// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ViewQuery Component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ViewQuery />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "params": undefined,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "params": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": false,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
